technique/muon/tests/test_muon_begin_end.py

```diff
@@ -0,0 +1,249 @@
+    def test_WHEN_set_users_to_non_empty_THEN_users_changed(self, _, __):
+        # Act
+        muon_begin_end.set_users()
+
+        # Assert
+        self.assertTrue(g.change_users.called, "Should have called to set new users")
+
+    @patch.object(g, "change_users")
+    @patch('technique.muon.muon_begin_end.get_input', return_value="")
+    def test_WHEN_set_geo_beamline_par_to_non_empty_THEN_beamline_par_changed(self, _, __):
+        # Act
+        muon_begin_end.set_users()
+
+        # Assert
+        self.assertFalse(g.change_users.called, "Should not have called to set new users")
+
+    @patch.object(g, "change_sample_par")
+    @patch('technique.muon.muon_begin_end.get_input', return_value="new comment")
+    def test_WHEN_set_comments_sample_par_to_non_empty_THEN_sample_par_changed(self, _, __):
+        # Act
+        old_sample_pars = {}
+        muon_begin_end.set_temp_sample_par(old_sample_pars)
+
+        # Assert
+        self.assertTrue(g.change_sample_par.called, "Should have called to set sample param")
+
+    @patch.object(g, "change_sample_par")
+    @patch('technique.muon.muon_begin_end.get_input', return_value="")
+    def test_WHEN_set_comments_sample_par_to_empty_THEN_sample_par_not_changed(self, _, __):
+        # Act
+        old_sample_pars = {}
+        muon_begin_end.set_temp_sample_par(old_sample_pars)
+
+        # Assert
+        self.assertFalse(g.change_sample_par.called, "Should not have called to set sample param")
+
+    @patch.object(muon_begin_end, "set_label")
+    @patch('technique.muon.muon_begin_end.get_input', side_effect=["n", "y"])
+    def test_WHEN_end_pre_cmd_called_label_incorrect_THEN_change_title_called(self, mock_set_label, _):
+        # Act
+        muon_begin_end.end_precmd()
+
+        # Assert
+        self.assertTrue(mock_set_label.called, "Should have called to set the label")
+
+    @patch.object(g, "change_title")
+    @patch('technique.muon.muon_begin_end.get_input', return_value="y")
+    def test_WHEN_end_pre_cmd_called_label_correct_THEN_change_title_not_called(self, _, __):
+        # Act
+        muon_begin_end.end_precmd()
+
+        # Assert
+        self.assertFalse(g.change_title.called, "Should not have called to set title")
+
+    def end_pre_cmd_get_input(self, input_prompt):
+        if input_prompt == "Is the run information correct (y/n)?":
+            self.get_input_call_count += 1
+            if self.get_input_call_count > 2:
+                return "y"
+        return "n"
+
+    @patch.object(muon_begin_end, "set_label")
+    def test_WHEN_end_pre_cmd_run_information_incorrect_twice_THEN_set_label_called_twice(self, set_label):
+        # Arrange
+        muon_begin_end.get_input = MagicMock(side_effect=self.end_pre_cmd_get_input)
+
+        # Act
+        muon_begin_end.end_precmd()
+
+        # Assert
+        self.assertEquals(set_label.call_count, 2, "Run information rejected twice so should have asked "
```

Comment:
```suggestion
        self.assertEqual(set_label.call_count, 2, "Run information rejected twice so should have asked "
```
`self.assertEquals` is depreciated. Instead use `self.assertEqual`.